SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs

```diff
@@ -112,43 +110,66 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
-                }
-                else {
-                    Competition dbCompetition;//assume the competition does not exist
-                    using (_context) {
-                        dbCompetition = _context.Competitions.Where(C => C.Id == Comp.Id).FirstOrDefault<Competition>();
-                        if (dbCompetition == null) {
-                            return BadRequest("Cannot update competition, no such competition!");
-                        }
-                        foreach (Requirement R in Comp.RequirementsList){
-                            Requirement TempRequirement = new Requirement {
-                                CompID = R.CompID,
-                                Standard = R.Standard,
-                                Accuracy = R.Accuracy,
-                                Total = R.Total
-                            };
-                            await _context.AddAsync(TempRequirement);
-                        }
-                        await _context.SaveChangesAsync();
-                        return Ok("All Requirements for the "+Comp.Name+" have been added");
+                if (requirements == null)
+                    return new BadRequestObjectResult("requirements cannot be null");
+                else
+                {
+                    //assume the competition does not exist
+                    Competition dbCompetition = _context.Competitions.Where(C => C.Id == CompID).FirstOrDefault<Competition>();
+                    if (dbCompetition == null)
+                    {
+                        return BadRequest("Cannot update requirements, competition does not exist!");
+                    }
+                    List<Requirement> requirementsList = GetRequirementsByID(CompID).ToList<Requirement>();
+                    for (int i =0; i < requirements.Length; i++)
+                    {
+                        Requirement R = requirementsList.ElementAt(i);
+                        R.Standard = requirements[i].Standard;
+                        R.Accuracy = requirements[i].Accuracy;
+                        R.Total = requirements[i].Total;
+                        _context.Requirements.Update(R);   
                     }
+                    await _context.SaveChangesAsync();
+                    return Ok("All Requirements for the " + dbCompetition.Name + " have been updated");
                 }
             }
-            else {
-                return new BadRequestObjectResult("Invalid competition model!");
+            else
+            {
+                return new BadRequestObjectResult("Invalid requirements model!");
             }
         }
-        public IEnumerable<Requirement> GetAllRequirementsByCompID(int CompID)
+ //##########################################################################################################
+        //GET: api/Competition/Requirements/{id}
+        /**returning an array of requirements for a given competition Id
```

Comment:
GC

Reply:
Well this is actually a description of what is happening. Not ghost code